### ISSUE TITLE
Model Tahminlerini Dışarı Aktarma ve İyileştirme

### ISSUE THREAD

**Issue by MesuttSahin** [MesuttSahin/teknofest_xray_sim#36]
###  BAĞIMLILIK DURUMU
* ** Girdi:** `models/` klasöründeki **En İyi Model** (`.pth`).
    * *Dikkat:* Loss grafiğinde Validation hatasının en düşük olduğu epoch dosyasını seç (Örn: `ep1.pth`). Son epoch genellikle ezbercidir (overfit), onu seçme.
* ** Çıktı:** Veri Analisti için `logs/predictions.npy` ve `logs/true_labels.npy`.

###  Görev Özeti
Analist, model başarısını ölçmek için ROC eğrisine ihtiyaç duyuyor. Bunun için modelin ürettiği ham sayıları (logits) değil, **olasılıkları (0-1 arası)** istiyor. Görevin `evaluate.py` dosyasını yazarak bu verileri dışarı aktarmak ve `transforms.py` dosyasını inceleyerek bir sonraki eğitim için plan yapmaktır.

---

####  BÖLÜM 1: KODLAMA 
** Çalışılacak Dosya:** `src/evaluate.py`

1.  **Modeli Yükle ve Hazırla:**
    * `best_model.pth` dosyasını yükle.
    * Modeli `.eval()` moduna al (Dropout ve BatchNorm kapansın).
    * **Kritik:** Validation Loader'ı `shuffle=False` olarak ayarla. (Karıştırırsan tahminler ve gerçekler eşleşmez!).

2.  **Döngü ve Sigmoid (Olasılık Hesabı):**
    * Döngü içinde validation setini gez.
    * `outputs = model(images)` -> (Bunlar Logits değerleridir, örn: -2.5, 4.1).
    * **ÖNEMLİ:** `probs = torch.sigmoid(outputs)` kodunu ekle.
    * *Açıklama:* Bu işlem çıktıları 0 ile 1 arasına sıkıştırır (Örn: %5, %98). Analist bunu bekliyor.
    * `probs` (Tahminler) ve `labels` (Gerçekler) değerlerini listelerde topla.

3.  **Veri İhracı (Numpy Save):**
    * Topladığın listeleri `numpy` dizisine çevir.
    * Dosyaları `logs/` klasörüne kaydet:
      ```python
      # Örnek Kayıt Kodu
      np.save('logs/predictions.npy', np.array(all_probs))
      np.save('logs/true_labels.npy', np.array(all_labels))
      print(" Tahminler Analist için hazırlandı.")
      ```

####  BÖLÜM 2: ARAŞTIRMA & PLANLAMA (Overfitting i Düzeltmek için)
** İncelenecek Dosya:** `src/data/transforms.py`
*(Burada kod yazma, sadece dosyayı incele ve not al)*

1.  **Durum Analizi:** Modelimiz muhtemelen veriyi ezberledi (Overfit oldu).
2.  **Reçete Hazırlığı:** 5. Gün yapacağımız eğitimde `src/data/transforms.py` dosyasını nasıl zorlaştırabiliriz?
    * *Araştır:* `RandomRotation` açısını artırmak? `ColorJitter` eklemek?
    * *Çıktı:* Bu önerilerini bir kenara not et, yarın uygulayacağız.

###  Checklist
- [ ] `evaluate.py` içinde `sigmoid` fonksiyonu kullanıldı mı?
- [ ] `evaluate.py` çalıştırıldı ve `logs/` klasöründe `.npy` dosyaları oluştu mu?
- [ ] `transforms.py` için iyileştirme önerileri not alındı mı?
